clients/adapter-amqp/src/main/java/org/eclipse/hono/adapter/client/command/amqp/ProtonBasedDeviceConnectionClient.java

```diff
@@ -122,41 +122,43 @@ protected void onDisconnect() {
 
     /**
      * {@inheritDoc}
+     * <p>
+     * Simply delegates to {@link #registerCommandConsumer(String, String, String, Duration, SpanContext)}.
      */
     @Override
-    public Future<JsonObject> getLastKnownGatewayForDevice(
+    public Future<Void> setCommandHandlingAdapterInstance(
             final String tenant,
             final String deviceId,
+            final String adapterInstanceId,
+            final Duration lifespan,
             final SpanContext context) {
 
-        Objects.requireNonNull(tenant);
-        Objects.requireNonNull(deviceId);
-
-        return getOrCreateDeviceConnectionClient(tenant)
-                .compose(client -> client.getLastKnownGatewayForDevice(deviceId, context));
+        return registerCommandConsumer(tenant, deviceId, adapterInstanceId, lifespan, context);
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
-    public Future<Void> setCommandHandlingAdapterInstance(
-            final String tenant,
+    public Future<Void> registerCommandConsumer(
+            final String tenantId,
             final String deviceId,
             final String adapterInstanceId,
             final Duration lifespan,
             final SpanContext context) {
 
-        Objects.requireNonNull(tenant);
+        Objects.requireNonNull(tenantId);
         Objects.requireNonNull(deviceId);
         Objects.requireNonNull(adapterInstanceId);
 
-        return getOrCreateDeviceConnectionClient(tenant)
+        return getOrCreateDeviceConnectionClient(tenantId)
                 .compose(client -> client.setCommandHandlingAdapterInstance(deviceId, adapterInstanceId, lifespan, context));
```

Comment:
FMPOV, an `UnsupportedOperationException` should be thrown here. The `registerCommandConsumer` method should not be invoked on the DeviceConnectionClient.

Same goes for the `unregisterCommandConsumer` method.